apps/emqx/src/emqx_persistent_session_ds/emqx_persistent_session_ds_node_heartbeat_worker.erl

```diff
@@ -0,0 +1,141 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023-2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+-module(emqx_persistent_session_ds_node_heartbeat_worker).
+
+-behaviour(gen_server).
+
+-include("session_internals.hrl").
+-include_lib("snabbkaffe/include/snabbkaffe.hrl").
+
+%% API
+-export([
+    create_tables/0,
+    start_link/0,
+    get_node_epoch_id/0,
+    get_last_alive_at/1
+]).
+
+%% `gen_server' API
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2
+]).
+
+%% call/cast/info records
+-record(update_last_alive_at, {}).
+
+-define(epoch_id_pt_key, {?MODULE, epoch_id}).
+-define(node_epoch, node_epoch).
+-define(tab, ?node_epoch).
+
+-record(?node_epoch, {
+    epoch_id :: reference(),
+    node :: node(),
+    last_alive_at :: pos_integer()
+}).
+
+-type epoch_id() :: reference().
+
+%%--------------------------------------------------------------------------------
+%% API
+%%--------------------------------------------------------------------------------
+
+-spec start_link() -> {ok, pid()}.
+start_link() ->
+    create_tables(),
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+-spec get_node_epoch_id() -> epoch_id().
```

Comment:
Can we read it mnesia as well? I think this token is not accessed too frequently to justify use of persistent term.

Reply:
Nevermind, the table is not keyed by node.